File: leet/reverse-integer.ts
/**
 * https://leetcode.com/problems/reverse-integer/description/
 * Given a signed 32-bit integer x, return x with its digits reversed. If reversing x causes the value to go outside the signed 32-bit integer range [-2^31, 2^31 - 1], then return 0.
 * Assume the environment does not allow you to store 64-bit integers (signed or unsigned).
 */

export function reverse(x: number): number {
  const s = x.toString();
  let result = '';
  if (s[0] === '-') {
    result += '-';
    for (let i = s.length - 1; i >= 1; i--) {
      result += s[i];
    }
  } else {
    for (let i = s.length - 1; i >= 0; i--) {
      result += s[i];
    }
  }
  const intResult = parseInt(result);
  const MAX_INT = 2 ** 31;
  if (intResult < -MAX_INT) return 0;
  if (intResult > MAX_INT - 1) return 0;
  return intResult;
}

File: returned/python/reverse_integer.py
# https://leetcode.com/problems/reverse-integer/description/
class Solution:
    def reverse(self, x: int) -> int:
        if(x < 0):
            res = int('-' + str(x)[1:][::-1])
            if res < (-2)**31:
                return 0
            else:
                return res
        else:
            res = int(str(x)[::-1])
            if res > 2**31 - 1:
                return 0
            else:
                return res
